Drive/rover_drive_system.hpp

```diff
@@ -170,6 +179,41 @@ class RoverDriveSystem
     }
   };
 
+double StateOfCharge_MAX(){
+  sjsu::lpc40xx::I2c & i2c = sjsu::lpc40xx::GetI2c<0>();
+  i2c.Initialize();
+  sjsu::lpc40xx::Gpio & alert_pin = sjsu::lpc40xx::GetGpio<0, 1>();
+  alert_pin.Initialize();
+  bool callback_was_called   = false;
+        InterruptCallback callback = [&callback_was_called]() {
+          callback_was_called = true;
+        };
+  uint8_t address = 0b0110110;
+
+  Max170343 *batterySensor = new Max170343(i2c, alert_pin, callback, address);
+
+  units::voltage::volt_t batteryVoltage = batterySensor -> GetVoltage();
+  double Voltage = batteryVoltage.to<double>();
+  return Voltage/3.3;
+
+}
+
+double StateOfCharge_LTC(){
+  
+  sjsu::lpc40xx::Gpio &gpio = sjsu::lpc40xx::GetGpio<0, 1>();
+  sjsu::GpioCounter counter(gpio, sjsu::Gpio::Edge::kRising);
+
+  sjsu::lpc40xx::Gpio &pol = sjsu::lpc40xx::GetGpio<2, 3>();
+  units::impedance::ohm_t resistance = 10'000_Ohm;
+
+  Ltc4150 *batterySensor = new Ltc4150(counter,pol,resistance);
+
+  units::charge::microampere_hour_t batteryCharge = batterySensor->GetCharge();
+  double Charge = batteryCharge.to<double>();
+  return Charge/100.0;
+
+}
+
```

Comment:
I think we should move state of charge to its own class. That way it removes a lot of this code that will be duplicated within Arm.

Reply:
Also on line 213 100.0 is hardcoded for now. We need to replace it with the maH rating of the battery we are using.